eth/beacon/state_machines/base.py

```diff
@@ -0,0 +1,185 @@
+                                chaindb: BaseBeaconChainDB,
+                                depth: int,
+                                start_slot: int) -> Iterable[BaseBeaconBlock]:
+        if last_block_hash == GENESIS_PARENT_HASH:
+            return
+
+        block = chaindb.get_block_by_hash(last_block_hash)
+
+        for _ in range(depth):
+            yield block
+            try:
+                block = chaindb.get_block_by_hash(block.parent_hash)
+            except (IndexError, BlockNotFound):
+                break
+            if block.slot_number < start_slot:
+                break
+
+    @property
+    def parent_block(self) -> BaseBeaconBlock:
+        return self.chaindb.get_block_by_hash(
+            self.block.parent_hash
+        )
+
+    #
+    # CrystallizedState
+    #
+    @property
+    def crystallized_state(self) -> CrystallizedState:
+        """
+        Returns the latest CrystallizedState.
+        """
+        if self._crytallized_state is None:
+            self._crytallized_state = self.chaindb.get_crystallized_state_by_root(
+                self.parent_block.crystallized_state_root
+            )
+        return self._crytallized_state
+
+    @classmethod
+    def get_crystallized_state_class(cls) -> Type[CrystallizedState]:
+        """
+        Return the :class:`~eth.beacon.types.crystallized_states.CrystallizedState` class that this
+        StateMachine uses for crystallized_state.
+        """
+        if cls.crystallized_state_class is None:
+            raise AttributeError("No `crystallized_state_class` has been set for this StateMachine")
+        else:
+            return cls.crystallized_state_class
+
+    #
+    # ActiveState
+    #
+    @property
+    def active_state(self) -> ActiveState:
+        """
+        Returns latest active state.
+
+        It was backed up per cycle. The latest ActiveState could be reproduced by
+        ``old_active_state`` and recent blocks.
+
+        NOTE: The following logic in beacon chain spec will be changed with the current spec.
+        """
+        if self._active_state is None:
+            # Reproduce ActiveState
+            old_active_state_root = self.chaindb.get_active_state_root_by_crystallized(
+                self.crystallized_state.hash
+            )
+            old_active_state = self.chaindb.get_active_state_by_root(old_active_state_root)
+            old_active_state_slot = self.crystallized_state.last_state_recalc
+
+            depth = self.config.CYCLE_LENGTH * 2
+            blocks = self.get_prev_blocks_to_slot(
+                self.parent_block.hash,
+                self.chaindb,
+                depth,
+                start_slot=old_active_state_slot
+            )
+            blocks = blocks[::-1]
+
+            self._active_state = self.get_active_state_class().from_old_active_and_blocks(
+                old_active_state,
+                blocks,
+                self.config.CYCLE_LENGTH * 2,
```

Comment:
It appears that this is the only value that is accessed in `self.config`.  Does it make sense to just have this be a class level variable?